core/eolearn/core/eodata.py

```diff
@@ -796,6 +803,49 @@ def _load_data(file_handler, is_pickle):
             eopatch[feature] = data
         return eopatch
 
+    @staticmethod
+    def walk(filesystem, patch_location):
+        """ Recursively reads a patch_location and returns yelds tuples of (feature_type, feature_name, file_path)
+        """
+        for path in filesystem.listdir(patch_location):
+            if '.' not in path:
+                subdir = fs.path.combine(patch_location, path)
+                for file in filesystem.listdir(subdir):
+                    yield path, file.split('.')[0], fs.path.combine(subdir, file)
+            else:
+                yield path.split('.')[0], None, fs.path.combine(patch_location, path)
+
+    @staticmethod
+    def walk_filtered(filesystem, patch_location, features):
+        """ Based on EOPatch.walk generator, but applies feature filtering based on the `features` argument
+        """
+        itr = EOPatch.walk(filesystem, patch_location)
+        itr = ((FeatureType(ftype), fname, path) for ftype, fname, path in itr)
+
+        features = FeatureParser(features).feature_collection
+        for ftype, fname, path in itr:
+            if ftype in [FeatureType.BBOX, FeatureType.TIMESTAMP, FeatureType.META_INFO]:
+                yield ftype, fname, path
+
+            if ftype not in features.keys():
+                continue
+
+            ftrs = features[ftype]
+            if ftrs == Ellipsis or fname in ftrs:
```

Comment:
It is better to check `ftrs is Ellipsis` instead of `ftrs == Ellipsis`

Reply:
Done.